JanggiGame.py

```diff
@@ -227,32 +238,24 @@ def get_general(self, color):
         with a specified color
         """
         general_name = color + "Gn"
-        board = self.get_board()
         # iterate through all pieces on the game board
-        for row in board:
-            for piece_obj in row:
-                # if not empty...
-                if piece_obj is not None:
-                    # if general, return general
-                    if general_name in piece_obj.get_name():
-                        return piece_obj
+        for piece_obj in self.get_all_pieces():
+            if general_name in piece_obj.get_name():
+                return piece_obj
 
     def all_player_moves(self, color):
         """
-        helper function returns a large list of all the possible moves a player (color)
+        helper function returns a large dictionary of all the possible moves a player (color)
         can make with their current set of pieces on the game board
+        key = piece's position
+        val = list of valid moves
         """
         all_valid_moves = dict()
-        board = self.get_board()
         # iterate through all pieces on the game board
-        for row in board:
-            for piece_obj in row:
-                # if not empty...
-                if piece_obj is not None:
-                    # if color
-                    if color in piece_obj.get_name():
-                        # get the valid moves and add them to the enemy list
-                        all_valid_moves[piece_obj.get_numeric_position()] = piece_obj.get_valid_moves()
+        for piece_obj in self.get_all_pieces():
+            if color in piece_obj.get_color():
```

Comment:
Could create another generator (that extends the first) to only return one player's pieces.

Reply:
Good idea! I'll try it out.